mods/beds/api.lua

```diff
@@ -96,9 +115,10 @@ function beds.register_bed(name, def)
 				return false
 			end
 			node.param2 = new_param2
-			minetest.swap_node(pos, node)
-			minetest.remove_node(p)
-			minetest.set_node(newp, {name = node.name:gsub("%_bottom", "_top"), param2 = new_param2})
+			-- do not remove_node here - it will trigger destroy_bed()
```

Comment:
It does (traced it down to ```environment.cpp```), but somehow this works. No matter how hard I try, calling ```remove_node``` messes it up.